server/commands/list_objects.go

```diff
@@ -94,13 +152,49 @@ func (q *ListObjectsQuery) Execute(ctx context.Context, req *openfgapb.ListObjec
 	}, nil
 }
 
-func (q *ListObjectsQuery) ExecuteStreamed(ctx context.Context, req *openfgapb.StreamedListObjectsRequest, srv openfgapb.OpenFGAService_StreamedListObjectsServer) error {
+func (q *ListObjectsQuery) ExecuteStreamed(
+	ctx context.Context,
+	req *openfgapb.StreamedListObjectsRequest,
+	srv openfgapb.OpenFGAService_StreamedListObjectsServer,
+) error {
 	err := q.validateInput(ctx, req.StoreId, req.Type, req.AuthorizationModelId, req.Relation)
 	if err != nil {
 		return err
 	}
 
-	resultsChan := make(chan string, q.ListObjectsMaxResults)
+	targetObjectType := req.GetType()
+	targetRelation := req.GetRelation()
+
+	model, err := q.Datastore.ReadAuthorizationModel(ctx, req.GetStoreId(), req.GetAuthorizationModelId())
+	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			return serverErrors.TypeNotFound(targetObjectType)
+		}
+		return err
+	}
+
+	typesys := typesystem.New(model)
+
+	_, ok := typesys.GetTypeDefinition(targetObjectType)
+	if !ok {
+		return serverErrors.TypeNotFound(targetObjectType)
+	}
+
+	relation, ok := typesys.GetRelation(targetObjectType, targetRelation)
+	if !ok {
+		return serverErrors.RelationNotFound(targetRelation, targetObjectType, nil)
+	}
+
+	hasTypeInfo, ok := typesys.HasTypeInfo(targetObjectType, targetRelation)
+	if !ok {
+		// todo(jon-whit): handle error
```

Comment:
🤔 wondering the best course of action here. If we've made it this far we damn well better be referencing a valid `targetObjectType` and `targetRelation`, and so it seems very unlikely we'd hit this case. I wonder if logging a warning is simply best here.

WDYT?